### PR TITLE
Build ARM64 image in Action

### LINKED ISSUE #26: ARM64 images
Hi, thanks for your work on this! Would you be open to modifying the Actions pipeline to push arm64 versions of this image to the registry as well?

### PR DESCRIPTION
**Pull request by jtbaker**
Resolves #26 (I think)

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -26,13 +26,15 @@ if [ ! -d graphhopper ]; then
...
 imagename="israelhikingmap/graphhopper:${1:-latest}"
+arm64imagename="israelhikingmap/graphhopper:arm64-${1:-latest}"
 if [ "$1" ]; then
   echo "Checking out graphhopper:$1"
   (cd graphhopper; git checkout --detach "$1")
 fi
 
 echo "Building docker image ${imagename}"
-docker build . -t ${imagename}
+docker buildx build --platform linux/amd64 . -t ${imagename}
```
Comment (HarelM) [inline build.sh:36]: I this correct?

**Comment by HarelM** [comment]
The build is falling, maybe due to tests.
I'll wait for it to be fixed before merging this...

File: build.sh (hunk 1 above)
Comment (jtbaker) [reply build.sh:36]: Using `docker buildx` seems to be the recommended pattern to do this when trying to build an image for a different architecture platform (GH actions) than the one that is doing the building (https://medium.com/swlh/using-github-actions-to-build-arm-based-docker-images-413a8d498ee). Where are you seeing the tests fail?

Comment (jtbaker) [reply build.sh]: Removed the `buildx` usage from the non-AMD64 version to see if that helps.

Comment (HarelM) [reply build.sh]: Do I need to install something special to run this?

Comment (jtbaker) [reply build.sh]: I think the first time you run it, you need to preface it with `docker buildx create <contextname>` per https://docs.docker.com/engine/reference/commandline/buildx_create/

Unfortunately I can't replicate on my end trying to build the `linux/amd64` version due to  some virtualization issues in QEMU :-(

Comment (HarelM) [reply build.sh]: If there's an extra setup needed please add the relevant instructions to the readme.

File: .github/build-and-upload.sh [hunk 2]
```diff
@@ -4,9 +4,11 @@ echo "Cloning graphhopper"
 git clone https://github.com/graphhopper/graphhopper.git
 echo "Building docker image"
 docker build . -t israelhikingmap/graphhopper:latest
+docker buildx build --platform linux/arm64 -f arm64.Dockerfile .  -t israelhikingmap/graphhopper:arm64-latest
```
Comment (HarelM) [inline .github/build-and-upload.sh:7]: What's the difference between the docker files? I didn't see something that requires a different file, did I miss anything?

**Comment by HarelM** [comment]
Also the build and build-and-publish are very similar, is there a way to consolidate them? maybe use the build as part of the build-and-publish?
@zstadler feel free to share your perspective.

**Comment by zstadler** [comment]
> Also the build and build-and-publish are very similar, is there a way to consolidate them? maybe use the build as part of the build-and-publish?

Yes, `build-and-publish.sh` can be modified to use `build.sh`. Also, `build.sh` can be modified to accept an optional `--publish` flag to further simplify its use by `build-and-publish.sh`.

File: .github/build-and-upload.sh (hunk 2 above)
Comment (jtbaker) [reply .github/build-and-upload.sh:7]: Different base images; The Maven-JDK image isn't published for the ARM64 image, so needed to update to `FROM arm64v8/maven:3.8.5-openjdk-18 as build`

**Comment by jtbaker** [comment]
Hi, sorry for the delay on responses to this PR. I've managed to personally bypass this issue by utilizing the new virtualization framework feature on Docker Desktop for Mac, allowing me to run images built for amd64 on my M1 chip more seamlessly as mentioned here: https://twitter.com/kelseyhightower/status/1601383577035235328

I'll take a look at your comments and see about updating my changes in the PR if you think it's worthwhile to try and get it in. I'm also happy to close the PR if you don't want to deal with it.

**Comment by HarelM** [comment]
If this is solved by Docker and the extra image is not really needed, I would prefer to close this PR.
Might be worth mentioning in the readme about how to solve this issue, or even link to your comment.

**Comment by Lermex** [comment]
hey @HarelM my company is using an ARM-based cluster on AWS so we do need an architecture-specific docker image because the virtualization feature is only available on macs. Would you mind reopening this? I can try to help with the remaining work to get the PR in a good shape if needed

**Comment by HarelM** [comment]
Simply create a new PR, no need to reopen this one...

**Comment by jtbaker** [comment]
@Lermex should have a version of this working now with the merge of #41

**Pull request closed without merging**